### PR TITLE
[CHE-120] Add Total Posts Display On Threads

### PR DESCRIPTION
**Pull request by howardsun-dev**
### Description
This PR aims to:

- add total post count on individual threads
- add total reply count per thread on threadDisplay for All
- add total reply count per thread on threadDisplay custom forums 

Extra:  
- add to queryHelper in `server/controllers/helpers/queryHelpers.ts` to handle MongoDb aggregate for sorting
- add type to forums for postCount

### Jira Task

[CHE-120](https://codehammers.atlassian.net/jira/software/projects/CHE/list?selectedIssue=CHE-120&text=CHE-120)

### Testing Instructions
- Ensure branches are up to date.
- Navigate to Forums and click through All Forums and each Forum (ex: General) to see if reply count appears on the screen for each individual thread.
- Click into the individual threads, and there should be a total post count for that thread (including the OP)

#### All Team Members

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [ ] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [x] I ran `npm run docker-test` in my local environment to check that this PR passes all unit tests.
- [x] I did a quick check to make sure my code changes follow the recomended style guide.

### Additional Notes, Images, etc.

(This is space for any additional information or attachments you'd like to add as part of this PR.)

[CHE-120]: https://codehammers.atlassian.net/browse/CHE-120?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/src/components/Forums/ThreadDetails/ThreadDetails.tsx [hunk 1]
```diff
@@ -96,6 +99,9 @@ const ThreadDetail = ({ forumId, threadId }: ThreadDetailProps) => {
           </div>
         ))}
       </div>
+      <div className="mt-4">
+        <strong>Total Posts:</strong> {totalPosts}
```
Comment (seantokuzo) [inline client/src/components/Forums/ThreadDetails/ThreadDetails.tsx]: From a UX perspective this feels a little weird seeing # of replies in ThreadDisplay and then # posts adds the original post to the count in this component. My vote would be either stick with # replies here or not have the count at all in this component

File: client/src/components/Forums/ThreadDetails/ThreadDetails.tsx [hunk 2]
```diff
@@ -59,6 +59,9 @@ const ThreadDetail = ({ forumId, threadId }: ThreadDetailProps) => {
     }
   };
 
+  const totalPosts: number = posts.length + 1;
+
+  if (pending) return null;
```
Comment (seantokuzo) [inline client/src/components/Forums/ThreadDetails/ThreadDetails.tsx:62]: the `pending` state seems to repeat the functionality of the `loading` state. Could we reuse the `loading` state in the handleDeletePost method or do we need the separate `pending` state?

File: client/src/components/Forums/ThreadDetails/ThreadDetails.tsx (hunk 1 above)
Comment (brok3turtl3) [reply client/src/components/Forums/ThreadDetails/ThreadDetails.tsx]: I agree. I think just having the totals show on the ThreadsDisplay page is sufficient. Once the user navigates inside the thread they are already looking at all the posts.

File: client/src/components/Forums/ThreadDetails/ThreadDetails.tsx (hunk 2 above)
Comment (brok3turtl3) [reply client/src/components/Forums/ThreadDetails/ThreadDetails.tsx:62]: @seantokuzo I see what you are saying for sure. They are two different situations so I do feel there is value in differentiating though. It was really a quick fix to make sure there was some state-handling logic while it awaits migration to redux where the reducer state will handle all of this and not feel so clunky. I think we can leave it for now.

File: client/src/components/Forums/ThreadsDisplay/ThreadsDisplay.tsx [hunk 3]
```diff
@@ -77,6 +83,8 @@ const ThreadsDisplay = ({ forumId, onThreadSelect }: ThreadsDisplayProps) => {
             <small>
               Started by {thread.user.firstName} on{' '}
               {new Date(thread.createdAt).toLocaleDateString()}
+              <span className="mx-2 text-gray-400">|</span>
+              <span className="italic">{formatReplies(thread.postCount)}</span>
```
Comment (seantokuzo) [inline client/src/components/Forums/ThreadsDisplay/ThreadsDisplay.tsx:87]: ⛏️ feel free to ignore this but I like declaring as few helper functions as possible to save from cluttering up the component. Personal preference tho

**Review by jimpulsee — APPROVED** [review]

File: server/controllers/helpers/queryHelpers.ts [hunk 4]
```diff
@@ -1,14 +1,81 @@
...
+export const aggregateSort = <T>(
+  pipeline: AggregateQuery<T>,
+  sortField: string = 'createdAt',
+  sortOrder: SortOrder = -1,
+) => {
+  const sortObj: Record<string, SortOrder> = { [sortField]: sortOrder };
+  return pipeline.sort(sortObj).exec();
+};
+
+export const aggregateThreadsWithPostCount = (
```
Comment (seantokuzo) [inline server/controllers/helpers/queryHelpers.ts:43]: Nice work with this helper! 🔥

**Review by seantokuzo — APPROVED** [review]
Left a few comments but this all is working great! Nice work Howard!

File: client/src/components/Forums/ThreadDetails/ThreadDetails.tsx (hunk 1 above)
Comment (howardsun-dev) [reply client/src/components/Forums/ThreadDetails/ThreadDetails.tsx]: Pushed an update to remove it

**Pull request merged**